Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1080,11 +1080,14 @@ void WebProcessProxy::createGPUProcessConnection(IPC::Connection::Handle&& conne
     parameters.ignoreInvalidMessageForTesting = ignoreInvalidMessageForTesting();
 #endif
     parameters.isLockdownModeEnabled = lockdownMode() == WebProcessProxy::LockdownMode::Enabled;
+    m_gpuProcessConnectionIdentifier = identifier;
```

Comment:
Can we assert that we don't already have an identifier?

It seems like sort of weird state to hold, since a WebProcess could in theory (but not in practice, currently) have multiple GPUProcessConnections.

Is there a guarantee that a GPUProcess crash would run `gpuProcessExited` in the UI process, before we receive a message from the WebProcess asking for a new connection (if that received the crash notification first)?

Reply:
> Can we assert that we don't already have an identifier?

Hmm, ok..

> It seems like sort of weird state to hold, since a WebProcess could in theory (but not in practice, currently) have multiple GPUProcessConnections.

We'd have list of known GPUProcessConnections. 

> Is there a guarantee that a GPUProcess crash would run `gpuProcessExited` in the UI process, before we receive a message from the WebProcess asking for a new connection (if that received the crash notification first)?

Yes, that's an existing race condition. The UI thinks the GPUP is still valid, and tries to send the new connections there.
However, nowadays the unresponsiveness notification should take care of it.